prow/spyglass/gcsartifact_test.go

```diff
@@ -0,0 +1,248 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package spyglass
+
+import (
+	"bytes"
+	"compress/gzip"
+	"context"
+	"io"
+	"testing"
+
+	"cloud.google.com/go/storage"
+	"k8s.io/test-infra/prow/spyglass/viewers"
+)
+
+type ByteReadCloser struct {
+	io.Reader
+}
+
+func (rc *ByteReadCloser) Close() error {
+	return nil
+}
+
+type fakeArtifactHandle struct {
+	oAttrs   *storage.ObjectAttrs
+	contents []byte
+}
+
+func (h *fakeArtifactHandle) Attrs(ctx context.Context) (*storage.ObjectAttrs, error) {
+	return h.oAttrs, nil
+}
+
+func (h *fakeArtifactHandle) NewRangeReader(ctx context.Context, offset, length int64) (artifactReader, error) {
+	toRead := length
+	var err error
+	lenContents := int64(len(h.contents))
+	if offset+length > lenContents {
+		toRead = lenContents - offset
+		err = io.EOF
+	}
+	return &ByteReadCloser{bytes.NewReader(h.contents[offset : offset+toRead])}, err
```

Comment:
Is this how ObjectHandle works? It will return a artifactReader + io.EOF if the requested length is longer?

Reply:
no it will return an HTTP error with the string "EOF" contained in it. I chose to return io.EOF because this does not make a network call

Reply:
It sounds like a problem if your range reader differs from the actual implementation in a meaningful way?

Reply:
Same here?